examples/llm/src/models/mosaic_gpt/configuration_mosaic_gpt.py

```diff
@@ -45,6 +46,9 @@ def __init__(
         embedding_fraction: float = 1.0,
         low_precision_layernorm: bool = False,
         use_cache: bool = True,
+        tensor_parallel_mlp: bool = False,
+        world_size: int = 1,
+        local_world_size: int = 1,
```

Comment:
Is there any reason these values cannot be calculated dynamically in the model or layer init?

If I share a model config with someone else, I don't think they need to know the dist information I used at training time, and in fact they may change it if they want to train/inference on a different set of machines. 

Tagging @vchiley because we have a similar issue with MoE layer init and I want to make sure we do the same thing in both places.

Reply:
I didn't want to add dist dependency on init. It may or may not be initialized at init. For example, as you mentioned, during inference it may not be available and having dist dependency breaks model init.

Reply:
composer dist should just return a default value (1) if dist is not initialized. does that make it ok to just do inside the model instead of as a config var?

Reply:
Yes. That should work.

Reply:
Removed these config params.